### ISSUE TITLE
onionshare-gui: Missing Application Icon and QR Code Issue

### ISSUE THREAD

**Issue by W-Y-R** [NixOS/nixpkgs#245721]
### Describe the bug
Upon installing the onionshare-gui package from nixpkgs, two significant issues arise: first, there is no application icon available, making access limited to the console only. Second, the QR code feature encounters problems and does not function as expected (**```TypeError: Image.__init__() got an unexpected keyword argument 'qrcode_modules'```**)

### Steps To Reproduce
1. Install the onionshare-gui package from nixpkgs.
2. Attempt to search for it in the application menu.

### Expected behavior
My expectation is that the onionshare-gui package should appear in the application menu, and the QR code feature should function correctly as intended.

### Notify maintainers
@lourkeur

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - host os: `Linux 6.4.4, NixOS, 23.11 (Tapir), 23.11pre507322.b12803b6d90`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixos"`
 - channels(user): `"home-manager-23.05.tar.gz"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by gp2112** [comment]
>  onionshare-gui package should appear in the application menu

Fixed in #245914.

>  the QR code feature should function correctly as intended.

This is actually a upstream problem. They have already fixed in a [commit](https://github.com/onionshare/onionshare/commit/70a703a98), but not released. Yet, I've added a patch in #246108 that fixes it

**Comment by gp2112** [comment]
Closing, since all fixes were merged
